analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,148 @@
+##########################################
+#Purpose: Code to analyze and plot ssGSEA Results
+#Author: Pichai Raman
+#Date: 9/23/2019
+##########################################
+
+
+#Call libraries
+library("tidyverse")
+library("pheatmap")
+library("RColorBrewer")
+library("viridis")
+
+#Read in data
+clinData <- read.delim("../../data/pbta-histologies.tsv", stringsAsFactors=F)
+geneSetExpMat <- readRDS("../../scratch/GeneSetExpressionMatrix.RDS")
+
+#Choose Clinical Samples corresponding to samples in Gene Set Matrix
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2]
+	return(c(tmp1, tmp2))
+}
+
+#Analyse Pathways statistically correlated with clinical variables
+#Returns ANOVA P-value and Tukey HSD Results
+#Requires Pathway (pathwayName) and Clinical Variable (clinVarName)
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar")
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
```

Comment:
Also don't understand what's going on here. Is there a way to access these fields by name instead of a series of indices?

Reply:
I updated some of them, not sure of the "named" way to pull the DF from TukeyHSD.  It probably exists but can't easily determine.